tests/test_load_results/test_mteb_results.py

```diff
@@ -160,42 +149,41 @@ def test_revision_layer(json_path, results_path):
         ], f"Unexpected path structure for version {json_version}: '{relative_path}'"
 
 
-@pytest.mark.xfail
+@pytest.mark.xfail(reason="PairClassification and Reranking fail that test")
```

Comment:
Since tasks type can have different main metrics I would instead check that the main metric of the task is present in its results.

```
results = mteb.load_results()
# results is a dict["model_name", dict["revision": MTEBResults(...)]
# this function loads the results from the results repo

for model, revscores in results.items():
  for rev, mtebscore in revscores.items():
    task = mteb.get_task(mtebscore.task_name)
    score = mtebscore.get_score(getter = lambda scores: scores[task.metadata.main_score])
    assert score == mtebscore.get_score(getter = lambda scores: scores["main_score"])
```

Reply:
This way as long as the main score is present we shouldn't have any issues.

Reply:
Your code tests the existing datasets, but what I've done is check whether the `main_metric` from each task is included in the task evaluation results in the current implementation. This test focuses on a slightly different aspect. I plan to add this code to a separate file, as it serves a distinct purpose from the existing tests.

Now PairClassifier will output `['similarity', 'cosine', 'manhattan', 'euclidean', 'dot', 'max', 'main_score']`, but multilingual PairClassification tasks currently have 'ap' as main metric. This test is focusing to test behavior like that

Reply:
I am not sure I get it. I don't see a case where this fails and I am sure that I want it to fail (and which wouldn't be caught by the above approach).

Reply:
When I run a task i get (ap included):

```
{
  "dataset_revision": "7335288588f14e5a687d97fc979194c2abe6f4e7",
  "evaluation_time": 5.820026159286499,
  "kg_co2_emissions": null,
  "mteb_version": "1.12.59",
  "scores": {
    "test": [
      {
        "cosine": {
          "accuracy": 0.6239067055393586,
          "accuracy_threshold": 0.774509072303772,
          "ap": 0.6484016684922852,
          "f1": 0.6789940828402367,
          "f1_threshold": 0.6167908310890198,
          "precision": 0.5510204081632653,
          "recall": 0.884393063583815
        }, 
        ...
        "hf_subset": "default",
        "languages": [
          "fas-Arab"
        ],
        "main_score": 0.6484016684922852,
        ...
        }
      }
    ]
  },
  "task_name": "FarsTail"
}
```
do you have cases where you don't get ap?

Reply:
but I def. see that it is not in the outermost layers (similar to how it used to be with STS). This seems like format to standardize.

Reply:
Yes it is appearing in "sublevel" of metrics, but I thought it should be in "top level" like in STS. Because previously there was 
```
cosine_pearson

Reply:
Completely agree, then we will get "cosine_ap", "manhattan_ap", "max_ap" etc., and convert the main metric to "max_ap". Will you create an issue on this? We will need to discuss the issue problem with a few of the others as well to make sure that we don't break compatibility with the existing leaderboard.